### ISSUE TITLE
make_folds is unnecessarily slow for big data and v-fold cross-validation

### ISSUE THREAD

**Issue by jeremyrcoyle** [tlverse/origami#20]
In the interest of generality, a `fold` is currently defined as training and validation index vectors. This allows some observations to be neither training nor validation, which is relevant for some cross-validation schemes, but not relevant for the most common v-fold case. This doesn't matter for small data, but for big sample sizes (say 1e6 observations), this takes a lot of time, and generates quite a bit more data than is really necessary to describe the folds in v-fold cross-validation.

Therefore, let's modify things so that a missing training index vector implicitly defines a training set as all observations not in the validation set. 

Let's also try to optimize `make_folds` for big data a bit. It's particularly slow when `cluster_ids` are specified. For example: `make_folds(cluster_ids=seq_len(1e6),V=10)`

**Comment by jeremyrcoyle** [comment]
This was fixed in https://github.com/tlverse/origami/pull/40
